os/io.c

```diff
@@ -928,20 +928,14 @@ FlushClient(ClientPtr who, OsCommPtr oc)
     if (FlushCallback)
         CallCallbacks(&FlushCallback, who);
 
-    size_t todo = notWritten; /* trying to write that much this time */
     while (notWritten) {
         errno = 0;
-        ssize_t len = _XSERVTransWrite(trans_conn, ((const char*)oco->buf) + written, todo);
+        ssize_t len = _XSERVTransWrite(trans_conn, ((const char*)oco->buf) + written, notWritten);
         if (len >= 0) {
             written += len;
             notWritten -= len;
-            todo = notWritten;
         }
-        else if (ETEST(errno)
-#ifdef EMSGSIZE                 /* check for another brain-damaged OS bug */
```

Comment:
Indeed that can only happen on datagram / message based sockets.

I'm not 100% certain on Windows Named Pipes (but for all I know we're also using them stream-based).
@jonhermansen can you confirm that ?

By the way, I am actually thinking of adding message based transports in the future, but that's still a long way to go, and most likely won't happen within xtrans, but OsComm.